Tests/StructuralTests/Differentiable2Tests.swift

```diff
@@ -36,10 +36,11 @@ struct LabeledPoint3 {
 extension LabeledPoint3: Structural {
     public typealias StructuralRepresentation =
         StructuralStruct<
+            LabeledPoint3,
             StructuralCons<
-                StructuralProperty<NoDerivative<String>>,
+                StructuralProperty<LabeledPoint3, String, NoDerivative<String>>,
                 StructuralCons<
-                    StructuralProperty<Point3>,
+                    StructuralProperty<LabeledPoint3, Point3, Point3>,
```

Comment:
Due to the fact that we don't hide property wrapper types, unlike KeyPaths we need to have 2 type parameters for value: one for wrapped and another for unwrapped value. This has unfortunate an unpleasant complexity cost.